src/mumble/GlobalShortcut_linux.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright 2005-2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#include "GlobalShortcut_linux.h"
+
+#include "Settings.h"
+
+#include <QtCore/QFileSystemWatcher>
+#include <QtCore/QSocketNotifier>
+
+#ifdef Q_OS_LINUX
+#	include <fcntl.h>
+#	include <linux/input.h>
+#endif
+
+// We define a global macro called 'g'. This can lead to issues when included code uses 'g' as a type or parameter name
+// (like protobuf 3.7 does). As such, for now, we have to make this our last incl
+#include "Global.h"
+
+// We have to use a global 'diagnostic ignored' pragmas because
+// we still support old versions of GCC. (FreeBSD 9.3 ships with GCC 4.2)
+#if defined(__GNUC__)
+// ScreenCount(...) and so on are macros that access the private structure and
+// cast their return value using old-style-casts. Hence we suppress these warnings
+// for this section of code.
+#	pragma GCC diagnostic ignored "-Wold-style-cast"
+// XKeycodeToKeysym is deprecated.
+// For backwards compatibility reasons we want to keep using the
+// old function as long as possible. The replacement function
+// XkbKeycodeToKeysym requires the XKB extension which isn't
+// guaranteed to be present.
+#	pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#endif
```

Comment:
I think we can remove that since FreeBSD 9.3 has gone EOL a long time by now (https://www.freebsd.org/releases/)